interactions/client/bot.py

```diff
@@ -390,10 +390,42 @@ async def _ready(self) -> None:
                 log.debug("Client is now ready.")
                 await self._login()
 
+    async def _stop(self) -> None:
+        """Stops the websocket connection gracefully."""
+
+        log.debug("Shutting down the client....")
+        self._websocket.ready.clear()  # Clears ready state.
+        self._websocket._closing_lock.set()  # Toggles the "ready-to-shutdown" state for the bot.
+        # And subsequently, the processes will close itself.
+
+        await self._http._req._session.close()  # Closes the HTTP session associated with the client.
+
     async def _login(self) -> None:
         """Makes a login with the Discord API."""
-        while not self._websocket._closed:
-            await self._websocket._establish_connection(self._shards, self._presence)
+
+        try:
+            await self._websocket.run()
+        except Exception as e:
+            log.exception(f"Websocket have raised an exception, closing because of: {e}")
```

Comment:
```suggestion
            log.exception("Websocket have raised an exception, closing because of:")
```
Doesn't this automatically print the full tb?